README.md

```diff
@@ -421,21 +421,56 @@ The `eval` function should be avoided.
 
 [⬆ back to TOC](#table-of-contents)
 
-##undefined
-Checking for `undefined` values should be done using the strict equality operator.
+## undefined
+
+Checking for `undefined` values should be done using the strict equality operator:
+
+> Explanation:
+> * In modern browsers (`IE9+`, `Opera 12.16+`, `Firefox 4+`) [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined)
+>   is immutable (a non-configurable, non-writable property of the global object).
+> * It prevents using variable before declaration.
 
 **Good:**
+
 ```js
-x === undefined;
+x === undefined
 ```
 
 **Bad:**
+
 ```js
 typeof x === 'undefined'
-
 x === void 0
 ```
 
+**Exceptions:**
+
+  * `typeof` should be used if you need to support old browsers (like `IE8`) where `window.undefined` property is mutable.
+  * `typeof` should be used for detect undeclared global variables in **different** environments:
+
+  ```js
+  // It works in browser environment only.
+  if (window.feature !== undefined) {
```

Comment:
There is my suggestion:

* If variable is declared and we want to check that is not `undefined` -> use equality with `undefined`
* If variable may not be declared -> use `typeof` or checking property of the global object.